### PR TITLE
[INFRA] Support for local paths Nate.infra.relative paths

### PR DESCRIPTION
**Pull request by NathanielRose**
- Tasks of https://github.com/microsoft/bedrock/issues/1149
- Documentation for relative paths
- Implementation for local paths
- Unit Tests

### CODE REVIEW

File: src/commands/infra/generate.ts [hunk 1]
```diff
@@ -501,6 +502,60 @@ export const singleDefinitionGeneration = async (
...
 
+/**
+ * Checks to see if module sources are local
+ *
+ * @param tfFile path to the terraform file in child directory
+ */
+export const checkModuleSource = (tfData: string): boolean => {
+  // Check if the file string matches an instance of a module source value as a local path
+  const matches = tfData.match(/^\s*source\s+=\s+["'](\.\.?\/[^"']*)["']$/gm);
+  if (matches != null) {
```
Comment (dennisseah) [inline src/commands/infra/generate.ts]: use `!==` and not `!=`

the code can be just
```
return matches !== null;
```

File: src/commands/infra/generate.ts [hunk 2]
```diff
@@ -501,6 +502,60 @@ export const singleDefinitionGeneration = async (
...
+      const moduleSource = new RegExp(splitLine[3].replace(/['"]+/g, ""), "g");
+      // Get relative path of terraform module local to the repo
+      const repoModulePath = await simpleGit(
+        path.join(
+          sourcePath,
+          fileSource.template,
+          splitLine[3].replace(/["']/g, "")
+        )
+      ).revparse(["--show-prefix"]);
+      // Concatendate the Git URL with munged data
```
Comment (dennisseah) [inline src/commands/infra/generate.ts]: concatenate

File: src/commands/infra/generate.ts [hunk 3]
```diff
@@ -576,6 +630,36 @@ export const generateConfig = async (
       templatePath
     );
   }
+  // TODO: Function to check for relative paths:
```
Comment (dennisseah) [inline src/commands/infra/generate.ts]: by adding this block of code into this `generateConfig` function shall make the function huge and you will find it hard to write unit test. I guess you move this block of code to a separate function

Comment (NathanielRose) [reply src/commands/infra/generate.ts]: Are you referring to lines 630 - 650? Im good with that -You're saying decouple from generateConfig and place the local path manipulation in a separate function.

Comment (dennisseah) [reply src/commands/infra/generate.ts]: yes: 630 - 654 to be precise. you can have these code in a new function and have `generateConfig` function to call this new function. The objective is to have smaller testable functions. thanks

File: src/commands/infra/generate.ts [hunk 4]
```diff
@@ -23,6 +23,7 @@ import { copyTfTemplate } from "./scaffold";
 import { build as buildError, log as logError } from "../../lib/errorBuilder";
 import { errorStatusCode } from "../../lib/errorStatusCode";
 import { exec } from "../../lib/shell";
+import { file } from "mock-fs";
```
Comment (dennisseah) [inline src/commands/infra/generate.ts]: i guess this is added by accident

File: src/commands/infra/generate.ts (hunk 3 above)
Comment (NathanielRose) [reply src/commands/infra/generate.ts]: Updated!

**Review by dennisseah — APPROVED** [review]

**Review by jmspring — APPROVED** [review]
Lgtm

File: src/commands/infra/generate.test.ts [hunk 5]
```diff
@@ -778,3 +801,79 @@ describe("Validate backend.tfvars file", () => {
     );
   });
 });
+
+describe("test checkModuleSource function", () => {
+  it("positive test", () => {
+    let res = checkModuleSource(`source="../../azure/aks-gitops"`);
+    expect(res).toBeTruthy();
```
Comment (yradsmikham) [inline src/commands/infra/generate.test.ts:808]: Let's change this to `expect(res).toBe(true)` instead. This article is good at explaining why: https://vincenttunru.com/toBeTruthy-vs-toBe-true/

Same for the one below.

**Review by yradsmikham — CHANGES_REQUESTED** [review]

**Comment by yradsmikham** [comment]
Also, did we need to update the smoke tests?

File: src/commands/infra/generate.test.ts (hunk 5 above)
Comment (dennisseah) [reply src/commands/infra/generate.test.ts:808]: checkModuleSource is returning boolean. using is toBeTruthy is ok.

**Comment by NathanielRose** [comment]
> Also, did we need to update the smoke tests?

This is in a separate PR

**Review by andrebriggs — APPROVED** [review]

**Review by yradsmikham — APPROVED** [review]

**Pull request merged**